lib/inline-registration.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const Plugin = require('broccoli-plugin');
+const fs = require('fs');
+const path = require('path');
+
+module.exports = class InlineRegistration extends Plugin {
+constructor(inputNodes, options) {
+    super(inputNodes, {
+      name: options && options.name,
+      annotation: options && options.annotation
+    });
+
+    this.options = options;
+  }
+
+  build() {
+    let indexHtml = fs.readFileSync(path.join(this.inputPaths[0], 'index.html')).toString();
+    let swRegistration = fs.readFileSync(path.join(this.inputPaths[1], 'sw-registration.js')).toString();
+
+    indexHtml = indexHtml.replace('ESW_INLINE_PLACEHOLDER', swRegistration);
+
+    fs.writeFileSync(path.join(this.outputPath, 'index.html'), indexHtml);
```

Comment:
This will force `index.html` to be re-rewitten for every rebuild. This breaks things like css hot-reloading (which is a built in feature of ember-cli) because it forces all rebuilds to modify the HTML itself (css hot-reloading requires a CSS file only change after the rebuild).

I'd suggest caching here to avoid writing the file on every rebuild (e.g. if the contents have not changed, don't write to it).

Reply:
Hi Rob, do you have a handy link demonstrating how to add caching to such a thing?